src/SmartEnum/SmartEnum.cs

```diff
@@ -287,27 +289,31 @@ public virtual bool Equals(SmartEnum<TEnum, TValue> other)
             return _value.Equals(other._value);
         }
 
+        /// <summary>
+        /// When this instance is one of the specified <see cref="SmartEnum{TEnum, TValue}"/> parameters.
+        /// Execute the action in the subsequent call to Then().
+        /// </summary>
+        /// <param name="smartEnumWhen">A collection of <see cref="SmartEnum{TEnum, TValue}"/> values to compare to this instance.</param>
+        /// <returns>A executor object to execute a supplied action.</returns>
+        public SmartEnumThen<TEnum, TValue> When(SmartEnum<TEnum, TValue> smartEnumWhen) =>
+            new SmartEnumThen<TEnum, TValue>(smartEnumWhen == this, false, this);
```

Comment:
Using `this.Equals(smartEnumWhen )` instead of `smartEnumWhen == this` saves the `null` check on `this` performed by the `==` operator.